### PR TITLE
#1635 use Carbon RadioButtonGroup in RadiosetControl instead of custom grouping code

### PR DESCRIPTION
**Pull request by veenas1**
Fixes : #1635 
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by matthoward366** [comment]
I think this either needs to wait until the unit tests are updated or as part of this PR get the unit tests working with RTL.

**Comment by nmgokhale** [comment]
@veenas1 This PR should be unblocked now. Jest tests are enabled in `main` branch.

**Comment by veenas1** [comment]
@matthoward366 Added a testcase, remaining tests are running fine. Could you please review.

**Review by matthoward366 — CHANGES_REQUESTED** [review]
Looks good.  One issue is the alignment of the tooltip is off.  This is caused by a margin added to the radio buttons.
Screenshot 2024-04-18 at 11 39 53 AM

File: canvas_modules/common-canvas/src/common-properties/controls/radioset/radioset.scss [hunk 1]
```diff
@@ -52,6 +40,9 @@
 			align-items: center;
 			margin: 2.5px 0 1.5px 0;
 		}
+		.tooltip-container {
+			margin-bottom: 3px; // align tooltip icon with the radio button
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/controls/radioset/radioset.scss:44]: Can we use the same spacing token used for the radio buttons?

**Comment by matthoward366** [comment]
With carbon 11 it looks like we can include the tooltip in the label itself.  Can you try that?  Then we don't need this special handling.

Screenshot 2024-04-18 at 4 35 52 PM

**Comment by veenas1** [comment]
> With carbon 11 it looks like we can include the tooltip in the label itself. Can you try that? Then we don't need this special handling.

`labelText` is the actual label for each radio button. There is another attr `helperText`, but seems like it's for whole radio group, not for single radiobutton.
image

**Comment by matthoward366** [comment]
>labelText is the actual label for each radio button

Right, but that label can also include the `Info` icon for each radio buttons description.  We've done that with other controls as well.

**Review by matthoward366 — APPROVED** [review]
Looks good

**Pull request merged**